R/create_sample_histograms.R

```diff
@@ -0,0 +1,84 @@
+#' Create a sample histogram from a population
+#'
+#' This function creates a grid of sample distributions from a population for different sample sizes.
+#' Author: Holly Williams
+#'
+#' @param pop the virtual population as a tibble
+#' @param samples the samples as a tibble
+#' @param var_name the name of the column for the variable that is being generated
+#' @param n_s a vector of the sample sizes
+#'
+#' @return a list of the sample histogram plots
+#' @export
+#'
+#' @examples
+#' pop <- generate_virtual_pop(100, height, rnorm, 0, 1)
+#' samples <- draw_samples(pop, 3, c(1, 10))
+#' create_sample_histograms(pop, samples, height, c(1, 10))
+create_sample_histograms <- function(pop, samples, var_name, n_s){
+
+  # START INPUT TESTS #
+  # convert var_name to string for testing
+  col_name <- toString(get_expr(enquo(var_name)))
+
+  # check pop df input
+  if (!is.data.frame(samples)) {
+    stop("'pop' should be input as a dataframe")
+  }
+  if (!is.element(col_name, colnames(samples))) {
+    stop(paste0("var_name (", col_name,") must be a column in 'pop' df"))
+  }
+
+  # check samples df input
+  if (!is.data.frame(samples)) {
+    stop("'samples' should be input as a dataframe")
+  }
+  if (!is.element(col_name, colnames(samples))) {
+    stop(paste0("var_name (", col_name,") must be a column in 'samples' df"))
+  }
+
+  # check n_s (number of samples )
+  if (!is.numeric(unlist(n_s)))
+    stop("sample sizes should be numeric")
+  for (i in n_s) {
+    if (i < 1)
+      stop("Samples sizes must be >=1")
+    if (!is.element(i, unique(samples$size)))
+      stop("Can only include sample sizes in 'size' column of 'samples' df")
+  }
+
+  # create an empty list of sample histograms
+  samples_hist <- list()
+
+  # for each sample size, generate tidy data needed for histogram plots
+  for (i in 1:length(n_s)){
+    samples_hist[[i]] <- samples %>%
```

Comment:
Please pay special attention to %>% as Tiffany mentioned during lecture

Reply:
Please see latest commit (I imported pipe operator)